### PR TITLE
Add integration test for search by catalog in user collection

### PR DESCRIPTION
**Pull request by mukeshk**
Addressed to #1097

### CODE REVIEW

**Comment by mukeshk** [comment]
The test cases which I have added works. 
When I ran the entire test suite, around 8 to 10 test cases broke down.
I need to investigate why the test cases are broken.

**Comment by mukeshk** [comment]
Create the WIP to discuss the issue. Thanks

**Comment by php-coder** [comment]
> When I ran the entire test suite, around 8 to 10 test cases broke down. I need to investigate why the test cases are broken.

I see that the tests don't fail on Travis CI, so it must be something related to your environment. If you provide more details, perhaps, I can help with debugging.

**Comment by php-coder** [comment]
@mukeshk Thank you for the PR! I'll review it during the day!

**Comment by mukeshk** [comment]
I had the application running in one console.
And in another console, I was running the robot framework test suite.
I ran it a couple of times without restarting the app server.

There were some JDBC exceptions for unique constraints.

**Comment by mukeshk** [comment]
Can you review ?

**Comment by php-coder** [comment]
> There were some JDBC exceptions for unique constraints.

Then, I'd say, that it works as designed. Some cases modify database and they can't be run more than once on the same instance of database. When we restart the app, the database is also get recreated.

File: src/test/robotframework/series/search/logic-user.robot [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Resource        ../../selenium.utils.robot
+Suite Setup     Before Test Suite
+Suite Teardown  Close Browser
+Force Tags      series  search  logic  user
+
+*** Test Cases ***
+Search series by non-existing catalog number in my collection
+	Go To                      ${SITE_URL}
+	Input Text                 id=catalogNumber  888
+	Select From List By Value  id=catalogName  michel
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: This test case should be similar to
https://github.com/php-coder/mystamps/blob/3914aa1fa34b9a2148ee99ae445248e6886dd15d/src/test/robotframework/series/search/logic.robot#L10-L14
with only exception that this time we search in user's collection

File: src/test/robotframework/series/search/logic-user.robot [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	Select Checkbox			   id=in-collection
+	Submit Form                id=search-series-form
+	Element Text Should Be     id=no-series-found  No series found
+	
+Search series by existing catalog number in my collection
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: This test case should be similar to
https://github.com/php-coder/mystamps/blob/3914aa1fa34b9a2148ee99ae445248e6886dd15d/src/test/robotframework/series/search/logic.robot#L16-L23
 with only exception that this time we search in user's collection

File: src/test/robotframework/series/search/logic-user.robot [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	Go To                        ${SITE_URL}
+	Input Text                   id=catalogNumber  99
+	Select From List By Value    id=catalogName  michel
+	Select Checkbox			     id=in-collection
+	Submit Form                  id=search-series-form
+	Page Should Contain Element  css=.search-results [href="/series/1"]
+
+*** Keywords ***
+Before Test Suite
+	Open Browser                        ${SITE_URL}/account/auth   ${BROWSER}
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: There should be only 2 spaces between `${SITE_URL}/account/auth` and `${BROWSER}` :-)

File: src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+	xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+		http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd">
+
+	<changeSet id="setup-test-search-in-my-collection" author="mukeshk" context="test-data">
+		<sqlFile path="classpath:/liquibase/sql/test-user-seriesowner-in-my-collection.sql" relativeToChangelogFile="false" />
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml]: Let's don't use `<sqlFile>` approach and follow the one that I've suggested already: https://github.com/php-coder/mystamps/issues/1097#issuecomment-517656016

In other words, we should use only liquibase instructions in order to populate data.

File: src/test/robotframework/series/search/logic-user.robot (hunk 1 above)
Comment (mukeshk) [reply src/test/robotframework/series/search/logic-user.robot]: ok.

File: src/test/robotframework/series/search/logic-user.robot (hunk 2 above)
Comment (mukeshk) [reply src/test/robotframework/series/search/logic-user.robot]: ok

File: src/test/robotframework/series/search/logic-user.robot (hunk 1 above)
Comment (mukeshk) [reply src/test/robotframework/series/search/logic-user.robot]: changed as suggested.

File: src/test/robotframework/series/search/logic-user.robot (hunk 2 above)
Comment (mukeshk) [reply src/test/robotframework/series/search/logic-user.robot]: changed as suggested.

**Comment by mukeshk** [comment]
comments addressed.

File: src/test/robotframework/series/search/logic-user.robot [hunk 5]
```diff
@@ -0,0 +1,41 @@
+*** Settings ***
+Documentation   Verify series search my collection scenarios
+Library         SeleniumLibrary
+Resource        ../../auth.steps.robot
+Resource        ../../selenium.utils.robot
+Suite Setup     Before Test Suite
+Suite Teardown  Close Browser
+Force Tags      series  search  logic  user
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: No, need "user" tag here.

File: src/test/robotframework/series/search/logic-user.robot [hunk 6]
```diff
@@ -0,0 +1,41 @@
+*** Settings ***
+Documentation   Verify series search my collection scenarios
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]:
```diff
-Verify series search my collection scenarios
+Verify scenarios of series search in user's collection
```

File: src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+			<column name="hash" value="$2a$10$8Rxlvw8r7r7a.w5rxOJYY.XbBE71ivvGjlnE6w/G73A58l1I76VRK" />
+			<column name="registered_at" valueComputed="${NOW}" />
+			<column name="activated_at" valueComputed="${NOW}" />
+		</insert>
+
+		<insert tableName="collections">
+			<column name="user_id" valueComputed="(SELECT id FROM users WHERE login = 'seriesowner')" />
+			<column name="slug" value="seriesowner" />
+			<column name="updated_at" valueComputed="${NOW}" />
+			<column name="updated_by" valueComputed="(SELECT id FROM users WHERE role = 'USER' ORDER by id LIMIT 1)" />
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml]: updated_by is the same as user_id typically:
https://github.com/php-coder/mystamps/blob/3914aa1fa34b9a2148ee99ae445248e6886dd15d/src/main/java/ru/mystamps/web/feature/collection/JdbcCollectionDao.java#L160-L165

Let's use the similar query for `updated_by` field then.

File: src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+		http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd">
+
+	<changeSet id="setup-test-search-in-my-collection" author="mukeshk" context="test-data">
+		<insert tableName="users">
+			<column name="login" value="seriesowner" />
+			<column name="role" value="USER" />
+			<column name="name" value="Series Owner" />
+			<column name="email" value="seriesowner@example.com" />
+			<column name="hash" value="$2a$10$8Rxlvw8r7r7a.w5rxOJYY.XbBE71ivvGjlnE6w/G73A58l1I76VRK" />
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml]: Could you also add a comment before this line:
```xml
<!-- Password: test -->
```

File: src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+			<column name="user_id" valueComputed="(SELECT id FROM users WHERE login = 'seriesowner')" />
+			<column name="slug" value="seriesowner" />
+			<column name="updated_at" valueComputed="${NOW}" />
+			<column name="updated_by" valueComputed="(SELECT id FROM users WHERE role = 'USER' ORDER by id LIMIT 1)" />
+
+		</insert>
+
+		<insert tableName="collections_series">
+			<column name="collection_id"
+					valueComputed="(SELECT c.id FROM collections c inner join users u  where u.login='seriesowner' and c.slug='seriesowner')" />
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml]: It seems like we can simplify the query to:

> SELECT id FROM collections WHERE slug='seriesowner'

File: src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml [hunk 10]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+	xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+		http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd">
+
+	<changeSet id="setup-test-search-in-my-collection" author="mukeshk" context="test-data">
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml]: It feels like this change set deserves a comment:
```xml
<comment>Creates the user "seriesowner" that has series #1 in its collection</comment>
```

File: src/test/robotframework/series/search/logic-user.robot [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+Documentation   Verify scenarios of series search in user's collection
+Library         SeleniumLibrary
+Resource        ../../auth.steps.robot
+Resource        ../../selenium.utils.robot
+Suite Setup     Before Test Suite
+Suite Teardown  Close Browser
+Force Tags      series  search  logic
+
+*** Test Cases ***
+Search series by non-existing catalog number in my collection
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: s/in my/in user's/

Comment (php-coder) [reply src/test/robotframework/series/search/logic-user.robot]: Even shorter: s/in my collection/in a collection/

File: src/test/robotframework/series/search/logic-user.robot [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	Go To                           ${SITE_URL}
+	Input Text                      id=catalogNumber  888
+	Select Random Option From List  id=catalogName
+	Select Checkbox			        id=in-collection
+	Submit Form                     id=search-series-form
+	Element Text Should Be          id=no-series-found  No series found
+
+Search series by existing catalog number in my collection
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: s/in my collection/in a collection/

File: src/test/robotframework/series/search/logic-user.robot [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+	solovyov    77
+	zagorski    83
+		
+*** Keywords ***
+Before Test Suite
+	Open Browser                        ${SITE_URL}/account/auth  ${BROWSER}
+	Register Keyword To Run On Failure  Log Source
+	Log In As                           login=seriesowner  password=test
+
+Search Series By Catalog Name And Number In My Collection
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: s/In My Collection/In Collection/

File: src/main/resources/liquibase/version/0.4.1.xml [hunk 14]
```diff
@@ -6,5 +6,6 @@
 		http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd">
 	
 	<include file="0.4.1/2019-07-21--series_sales_transaction_url_length.xml" relativeToChangelogFile="true"/>
+	<include file="0.4.1/2019-08-04--test_search_in_my_collection.xml" relativeToChangelogFile="true"/>
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1.xml]: 2019-08-04--test_search_in_my_collection.xml -> 2019-08-06--test_user_with_series_in_collection.xml

File: src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml (hunk 10 above)
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-04--test_search_in_my_collection.xml]: setup-test-search-in-my-collection -> create-a-user-with-a-series-in-his-collection

**Comment by mukeshk** [comment]
address the comments

File: src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+			<column name="registered_at" valueComputed="${NOW}" />
+			<column name="activated_at" valueComputed="${NOW}" />
+		</insert>
+
+		<insert tableName="collections">
+			<column name="user_id" valueComputed="(SELECT id FROM users WHERE login = 'seriesowner')" />
+			<column name="slug" value="seriesowner" />
+			<column name="updated_at" valueComputed="${NOW}" />
+			<column name="updated_by" valueComputed="(SELECT id FROM users WHERE login = 'seriesowner')" />
+
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml]: It's better to remove this extra line.

File: src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+		</insert>
+
+		<insert tableName="collections_series">
+			<column name="collection_id"
+					valueComputed="(SELECT id FROM collections where slug='seriesowner')" />
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml]: s/where/WHERE/

File: src/test/robotframework/series/search/logic-user.robot [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+
+Search series by existing catalog number in a collection
+	[Template]  Search Series By Catalog Name And Number In Collection
+	michel      99
+	scott       99
+	yvert       99
+	gibbons     99
+	solovyov    77
+	zagorski    83
+
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: Please, remove trailing spaces on this line.

File: src/test/robotframework/series/search/logic-user.robot [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+	Open Browser                        ${SITE_URL}/account/auth  ${BROWSER}
+	Register Keyword To Run On Failure  Log Source
+	Log In As                           login=seriesowner  password=test
+
+Search Series By Catalog Name And Number In Collection
+	[Arguments]                  ${name}  ${number}
+	Go To                        ${SITE_URL}
+	Input Text                   id=catalogNumber  ${number}
+	Select From List By Value    id=catalogName    ${name}
+	Select Checkbox			     id=in-collection
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: It looks like this line contains tabs between keyword and arguments. Please, use spaced for that.

File: src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml (hunk 16 above)
Comment (php-coder) [reply src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml]: P.S. it's ok to write it as a single line by the way.

File: src/test/robotframework/series/search/logic-user.robot [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+Library         SeleniumLibrary
+Resource        ../../auth.steps.robot
+Resource        ../../selenium.utils.robot
+Suite Setup     Before Test Suite
+Suite Teardown  Close Browser
+Force Tags      series  search  logic
+
+*** Test Cases ***
+Search series by non-existing catalog number in a collection
+	Go To                           ${SITE_URL}
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic-user.robot]: Let's move this `Go To` to the `Before Test Suite`. It should be last keyword, obviously.

File: src/main/java/ru/mystamps/web/support/togglz/Features.java [hunk 20]
```diff
@@ -54,6 +54,7 @@
 	SHOW_SUGGESTION_LINK,
 	
 	@Label("/site/index: search by catalog in collection")
+	@EnabledByDefault
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/support/togglz/Features.java]: As enabling of a new feature shouldn't be part of a commit with tests, I've extracted this line into a single commit and pushed it to master already: 8fedc5e57f7e63593812484d5cd503fa0a05df82 Also, I've updated changelog and has mentioned your contribution. Thank you!! 🥇 

So, could you, please, update your master and after that rebase this PR on top of updated master? :)

**Comment by php-coder** [comment]
Could you, please, modify commit message to be:
```
test(series/search/logic-user.robot): add tests for searching the series in user's collection.

Related to #673
Fix #1097
```

**Comment by php-coder** [comment]
Could you also remove the following comment?

https://github.com/php-coder/mystamps/blob/bea703f904810ae15b92a8d3313af8bf205740df/src/main/webapp/WEB-INF/views/site/index.html#L185

**Comment by mukeshk** [comment]
removed the todo

**Comment by php-coder** [comment]
It was the 8th issue that you closed! Thank you! Awesome! 👏

File: src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+
+		<insert tableName="collections">
+			<column name="user_id" valueComputed="(SELECT id FROM users WHERE login = 'seriesowner')" />
+			<column name="slug" value="seriesowner" />
+			<column name="updated_at" valueComputed="${NOW}" />
+			<column name="updated_by" valueComputed="(SELECT id FROM users WHERE login = 'seriesowner')" />
+		</insert>
+
+		<insert tableName="collections_series">
+			<column name="collection_id" valueComputed="(SELECT id FROM collections WHERE slug='seriesowner')" />
```
Comment (php-coder) [inline src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml]: There was a tiny formatting issue that I fixed in 76e636e4812035cc80b9454979c264bd53b9b51e Ideally, it's better to follow a similar approach in formatting everywhere, especially within a single file.

Comment (php-coder) [reply src/main/resources/liquibase/version/0.4.1/2019-08-06--test_user_with_series_in_collection.xml]: @mukeshk ^^^ JFYI

**Comment by php-coder** [comment]
P.S. By the way, don't forget to remove merged branches -- https://github.com/php-coder/mystamps/wiki/after_merge_steps#remove-merged-branch

**Pull request merged**